cmd/root.go

```diff
@@ -171,6 +177,8 @@ func init() {
 	rootCmd.Flags().StringVar(&buildJobSecurityContextConstraints, "build-job-security-context-constraints", "", "Run builds jobs using a specified SCC")
 	rootCmd.Flags().BoolVar(&buildJobGrantFullPrivilege, "build-job-full-privilege", false, "Run builds jobs using a privileged root user")
 	rootCmd.Flags().StringVar(&buildAdvancedConfigFilename, "build-job-advanced-config", "", "Add volumes, volume mounts and environment variables to your build jobs using a JSON file")
+	rootCmd.Flags().DurationVar(&gcInterval, "gc-interval", 30*time.Minute, "Run ContainerImageBuild cleanup operation according to this interval. Set to 0 to disable")
+	rootCmd.Flags().IntVar(&gcMaxKeepCount, "gc-max-keep", 5, "Delete all ContainerImageBuild resources in a 'finished' state that exceed this count")
```

Comment:
I actually don't think we've spoken about this, but I removed the TTL on the job in https://github.com/dominodatalab/forge/commit/23541dd3563e8d64d03fa9403aced3ba20f2847a#diff-dd01fb2f9549977a5fa7fb3c85839ec7L277. Mostly because an hour is quite low, but I do wonder if we should set a high one (like 1 day or something) or if we need something like this to clean those up.

Reply:
The question that comes immediately to mind is, "What does a terminated Pod cost?". I don't think it's much beyond a minor footprint of metadata storage in etcd; I doubt K8s keeps the Pod's former resource allocation locked up but I don't know for sure.

Obviously, when we nuke a `ContainerImageBuild` the owned `Job` and, by extension, the `Pod` the job owns will be deleted. We probably don't need this field and I added it just because.

It might also be worth removing this setting as well. The TTLController feature is in _alpha_, requires a gate setting to enable it right now, and, we should probably limit our delete process to using the owner `ContainerImageBuild` resource for consistency's sake, no?

https://github.com/dominodatalab/forge/blob/a87afe1dcf1a4f903cd5d1286f1552c8f0ad5ea0/controllers/containerimagebuild_resources.go#L239

Reply:
Oh interesting. Yeah it seems like we should operate on CIBs not jobs.

The secondary point is mostly around whether we should handle a potential CIB that is still running after some time that we deem excessive. i.e. if we have a build that just executes `sleep infinity` is that reasonable? Probably, since it's on the customer to figure out why they would do that. Or, we can set an upper limit and forcefully clean them up here.

As I was writing this however, I think I talked myself out of any sort of "active deadline" default at all for running CIBs.

Reply:
I agree, clients should handle user garbage like this. I think our responsibility is to prevent the accumulation of _finished_ cruft, not assume long-running builds are invalid and nuke them.